io-ballerina/build.gradle

```diff
@@ -109,111 +112,112 @@ task initializeVariables {
 
     gradle.taskGraph.whenReady { graph ->
         if (graph.hasTask(":${packageName}-ballerina:build") || graph.hasTask(":${packageName}-ballerina:publish") ||
-                graph.hasTask(":${packageName}-ballerina:publishToMavenLocal"))  {
-            ballerinaTest.enabled = false
+                graph.hasTask(":${packageName}-ballerina:publishToMavenLocal")) {
+            needSeparateTest = false
+
+            if (graph.hasTask(":${packageName}-ballerina:build")) {
+                needBuildWithTest = true
+            }
+            if (graph.hasTask(":${packageName}-ballerina:publish")) {
+                needBuildWithTest = true
+                needPublishToCentral = true
+            }
+            if (graph.hasTask(":${packageName}-ballerina:publishToMavenLocal")) {
+                needBuildWithTest = true
+            }
         } else {
```

Comment:
This will always set the `needSeparateTest` to true if we don't have `publishToMavenLocal` right?

Reply:
No, `needSeparateTest` becomes true when we don't do the following things:
1. Build
2. Publish
3. Publish to local maven